.github/workflows/pythonapp.yml

```diff
@@ -67,17 +64,70 @@ jobs:
+    # Getting TVM requires: 1) fetching TVM from github, 2) get LLVM, 3) cmake, 4) make, 5) install python dependecy.
+    # 1 to 4 will be retrieved from the cache.
+    # The pipeline only works for Unix systems. For windows we will have to compile LLVM from source which is a no go.
+    - name: Fetch and prepare TVM for compilation
+      if: ${{ steps.cache.outputs.cache-hit != 'true' && matrix.python-version != '3.5' && matrix.os != 'windows-2019' }}
+      run: |
+        cd ~/
+        git clone https://github.com/apache/incubator-tvm.git
+        cd incubator-tvm
+        git checkout tags/v0.7.0
+        git submodule update --recursive --init
+        cmake -E make_directory build
+    - name: Get LLVM on Linux
+      if: ${{ steps.cache.outputs.cache-hit != 'true' && matrix.python-version != '3.5' && matrix.os == 'ubuntu-latest' }}
+      working-directory: ../../../incubator-tvm
+      run: |
+        wget https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/clang+llvm-10.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz
+        tar -xf clang+llvm-10.0.0-x86_64-linux-gnu-ubuntu-18.04.tar.xz && mv clang+llvm-10.0.0-x86_64-linux-gnu-ubuntu-18.04 llvm
+    - name: Get LLVM on Mac
+      if: ${{ steps.cache.outputs.cache-hit != 'true' && matrix.python-version != '3.5' && matrix.os == 'macos-latest' }}
+      working-directory: ../../../incubator-tvm
+      run: |
+        wget https://github.com/llvm/llvm-project/releases/download/llvmorg-10.0.0/clang+llvm-10.0.0-x86_64-apple-darwin.tar.xz
+        tar -xf clang+llvm-10.0.0-x86_64-apple-darwin.tar.xz && mv clang+llvm-10.0.0-x86_64-apple-darwin llvm
+    - name: CMake TVM
+      if: ${{ steps.cache.outputs.cache-hit != 'true' && matrix.python-version != '3.5' && matrix.os != 'windows-2019' }}
+      working-directory: ../../../incubator-tvm/build
+      run: >-
+        cmake
+        "-DUSE_RPC=ON"
+        "-DUSE_GRAPH_RUNTIME=ON"
+        "-DUSE_LLVM=../llvm/bin/llvm-config"
+        ..
+    - name: Build TVM
+      if: ${{ steps.cache.outputs.cache-hit != 'true' && matrix.python-version != '3.5' && matrix.os != 'windows-2019' }}
+      working-directory: ../../../incubator-tvm/build
+      run: |
+        make -j3
+    - name: Install python TVM
+      if: ${{ matrix.python-version != '3.5' && matrix.os != 'windows-2019' }}
+      working-directory: ../../../incubator-tvm/python
+      run: |
+        python setup.py install
     - name: Lint with flake8
       run: |
         # stop the build if there are Python syntax errors or undefined names
         flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
         # The GitHub editor is 127 chars wide
         flake8 . --count  --max-complexity=10 --max-line-length=127 --statistics
+    # We don't run pytest for Linux py3.7 since we do coverage for that case.
     - name: Test with pytest
-      run: |
-        pytest
+      if: ${{ matrix.python-version != '3.7' && matrix.os != 'ubuntu-latest' }}
```

Comment:
Looking at the [CI run](https://github.com/microsoft/hummingbird/actions/runs/338421930), it looks like ALL of the py3.7 runs (including macOS py3.7/windows py3.7) skip this `Test with pytest` phase.  

The logic _looks_ right, but maybe it is being parsed wrong.

Reply:
Good catch! Let me try to rerun it.

Reply:
Even if I rerun it is still skipping `pytest`. No idea what is going on.